lms/views/basic_lti_launch.py

```diff
@@ -135,10 +135,24 @@ def unconfigured_basic_lti_launch(self):
         we'll save it in our DB. Subsequent launches of the same assignment
         will then be DB-configured launches rather than unconfigured.
         """
+        oauth_consumer_key = self.request.lti_user.oauth_consumer_key
+
+        lms_url = self.request.params.get("custom_canvas_api_domain")
+
+        if not lms_url:
+            lms_url = self.request.find_service(name="ai_getter").lms_url(
+                oauth_consumer_key
+            )
```

Comment:
This is the same as what the legacy views do -- they use the `custom_canvas_api_domain` launch param or, if that param is missing (e.g. if we're in a non-Canvas LMS) they fall back on the application instance's `lms_url` which is an almost completely unvalidated thing that the user typed in to the `/welcome` form. (The lack of validation of that form field is a problem in all sorts of ways and needs to be fixed later.)

Reply:
https://github.com/hypothesis/lms/issues/352